app/controllers/admin/reports_controller.rb

```diff
@@ -59,24 +60,30 @@ def render_report_as(format)
       if OpenFoodNetwork::FeatureToggle.enabled?(:background_reports, spree_current_user)
         filename = report_filename
         filename = "#{filename}html" if report_format.blank?
-        blob = ReportBlob.create_for_upload_later!(filename)
+        @blob = ReportBlob.create_for_upload_later!(filename)
         ReportJob.perform_later(
-          report_class, spree_current_user, params, format, blob
+          report_class, spree_current_user, params, format, @blob
         )
         Timeout.timeout(max_wait_time) do
-          sleep 1 until blob.content_stored?
+          sleep 1 until @blob.content_stored?
         end
 
         # This result has been rendered by Rails in safe mode already.
-        blob.result.html_safe # rubocop:disable Rails/OutputSafety
+        @blob.result.html_safe # rubocop:disable Rails/OutputSafety
       else
         @report.render_as(format)
       end
     end
 
     def render_timeout_error
       assign_view_data
-      @error = ".report_taking_longer"
+      if @blob
+        @error = ".report_taking_longer_html"
+        @error_url = @blob.url
+      else
+        @error = ".report_taking_longer"
+        @error_url = ""
+      end
```

Comment:
In what circumstances will `@blob` be empty? From what I can see, it will always be assigned before the timeout block is entered. So I think an empty blob would be an exception.

Reply:
When the feature toggle is off, we don't have a blob at all. Or if we time out before the blob is even created. That's highly unlikely but it could happen when the server is overloaded and there's maybe another callback first that takes longer. Or the report parameters do some queries for the filter inputs and in theory they could time out. But that's theoretical, it's really only needed when the feature is off. And then nginx cuts out before we see this error. This is just preventing a Bugsnag entry at the moment.

Reply:
> When the feature toggle is off, we don't have a blob at all.

Oh, I thought `Timeout::Error` would only be raised by the `Timeout` block which is inside the feature toggle.


> Or if we time out before the blob is even created.

But `@blob` is assigned before the `Timeout` block.

Anyway, it's all good :)